packages/cli/lib/lib/sw.js

```diff
@@ -0,0 +1,54 @@
+self.__precacheManifest = [].concat(self.__precacheManifest || []);
+
+/* global workbox */
+workbox.precaching.suppressWarnings();
+/** We are sure brotli is enabled for browsers supporting script type=module
+ * so we do brotli support only for them.
+ * We can do brolti support for other browsers but there is no good way of
+ * feature detect the same at the time of pre-caching.
+ */
+if (process.env.ENABLE_BROTLI && process.env.ES_BUILD) {
+  // Alter the precache manifest to precache brotli files instead of gzip files.
+  self.__precacheManifest = self.__precacheManifest.map(asset => {
+    if (/.*.js$/.test(asset.url)) {
+      asset.url = asset.url.replace(/.esm.js$/, ".esm.js.br");
+    }
+    return asset;
+  });
+
+  class BrotliRedirectPlugin {
+    // Before saving the response in cache, we need to treat the headers.
+    async cacheWillUpdate({request, response}) {
+      if (/.js.br$/.test(request.url)) {
+        const headers = response.headers;
```

Comment:
Would this work instead of manually cloning?

```js
response = response.clone();
response.headers.set('content-type', 'application/javascript')
return response;
```

Reply:
This should – it's what my service workers are doing, tho I just pray that they work

Reply:
my bad - this would work though:

```js
const headers = new Headers(clonedResponse.headers);
newHeaders.set('content-type', 'application/javascript');
const newResponse = new Response(await clonedResponse.text(), { headers })
```

Reply:
fixed